magenta/models/rl_tuner/rl_tuner.py

```diff
@@ -0,0 +1,2052 @@
+      # the target_network_update_rate.
+      self.target_network_update = []
+      for v_source, v_target in zip(self.q_network.variables(),
+                                    self.target_q_network.variables()):
+        # Equivalent to target = (1-alpha) * target + alpha * source
+        update_op = v_target.assign_sub(self.target_network_update_rate *
+                                        (v_target - v_source))
+        self.target_network_update.append(update_op)
+      self.target_network_update = tf.group(*self.target_network_update)
+
+    tf.scalar_summary('prediction_error', self.prediction_error)
+
+    self.summarize = tf.merge_all_summaries()
+    self.no_op1 = tf.no_op()
+
+  def train(self, num_steps=10000, exploration_period=5000, enable_random=True, verbose=False):
+    """Main training function that allows model to act, collects reward, trains.
+
+    Iterates a number of times, getting the model to act each time, saving the
+    experience, and performing backprop.
+
+    Args:
+      num_steps: The number of training steps to execute.
+      exploration_period: The number of steps over which the probability of
+        exploring (taking a random action) is annealed from 1.0 to the model's
+        random_action_probability.
+      enable_random: If False, the model will not be able to act randomly /
+        explore.
+      verbose: If True, will output debugging statements
+    """
+    print "Evaluating initial model..."
+    self.evaluate_model()
+
+    self.actions_executed_so_far = 0
+
+    if self.stochastic_observations:
+      tf.logging.info('Using stochastic environment')
+
+    self.reset_composition()
+    last_observation = self.prime_internal_models(suppress_output=False)
+
+    for i in range(num_steps):
+      # Experiencing observation, state, action, reward, new observation,
+      # new state tuples, and storing them.
+      state = np.array(self.q_network.state_value).flatten()
+      reward_rnn_state = np.array(self.reward_rnn.state_value).flatten()
+
+      if self.exploration_mode == 'boltzmann' or self.stochastic_observations:
+        action, new_observation, reward_scores = self.action(last_observation,
+                                                             exploration_period,
+                                                             enable_random=enable_random,
+                                                             sample_next_obs=True)
+      else:
+        action, reward_scores = self.action(last_observation,
+                                            exploration_period,
+                                            enable_random=enable_random,
+                                            sample_next_obs=False)
+        new_observation = action
+      new_state = np.array(self.q_network.state_value).flatten()
+      new_reward_state = np.array(self.reward_rnn.state_value).flatten()
+
+      if verbose:
```

Comment:
instead of a verbose flag and print statements, just use tf.logging and a lower log level. In this case, because these log statements actually take something to compute, you can include an if statement that's something like `if tf.logging.get_verbosity() <= tf.logging.DEBUG:`